lib/cc/analyzer/include_paths_builder.rb

```diff
@@ -0,0 +1,140 @@
+module CC
+  module Analyzer
+    class IncludePathsBuilder
+      def initialize(cc_excludes)
+        @cc_excludes = cc_excludes
+      end
+
+      def build
+        root = Directory.new('.', included_files)
+        root.included_paths
+      end
+
+      protected
+
+      def ignored_files
+        Tempfile.open(".cc_gitignore") do |tmp|
+          tmp.write(File.read(".gitignore")) if File.exist?(".gitignore")
+          tmp << @cc_excludes.join("\n")
+          tmp.close
+          tracked_and_ignored = `git ls-files -zi -X #{tmp.path}`.split("\0")
+          untracked_and_ignored = `git ls-files -zio -X #{tmp.path}`.split("\0")
+          tracked_and_ignored + untracked_and_ignored
+        end
+      end
+
+      def included_files
+        includable_files - ignored_files
+      end
+
+      def includable_files
+        tracked_in_git = `git ls-files -z`.split("\0")
+        untracked_in_git = `git ls-files -zo`.split("\0")
+        tracked_in_git + untracked_in_git
```

Comment:
My understanding is that the two sets are always mutually exclusive. `git ls-files` only shows tracked files and `git ls-files -o` only shows untracked files.